### ISSUE TITLE
Add integration with 'tasty'

### ISSUE THREAD

**Issue by chshersh** [sheyll/type-spec#2]
I would really appreciate integration of this amazing `type-spec` library with `tasty` testing framework. From [this issue](https://github.com/feuerbach/tasty/issues/199#issuecomment-355686089) I discovered that somebody needs to implement _tasty provider_ as a third-party package (probably called `tasty-type-spec`).

Currently it's not convenient enough to use `tasty` alongside with `type-spec`. You can see how awkward integration can be in this example:

* https://github.com/serokell/o-clock/blob/ee092f3d344a4cfe1f23e2d0ee99a2f2331f6721/test/Test/Time/TypeSpec.hs

Using `tasty-discovery` can really help a lot this test-suite!

Maybe you're interested in implementing this integration or at least in having it?

**Comment by sheyll** [comment]
Yeah, thanks for the interest. I will look into this, though I have no clue about tasty..(yet)
